_posts/2020-10-29-Rails-6-1-beta1-release.markdown

```diff
@@ -0,0 +1,71 @@
+---
+layout: post
+title: 'Rails 6.1: Horizontal Sharding, Multi-DB Improvements, Strict Loading, Destroy Associations in Background, Error Objects, and more!'
+categories: releases
+author: eileencodes
+published: true
+date: 2020-10-29 10:00:00 -05:00
+---
+The first beta for Rails 6.1 has been released and wow does it have a lot of great stuff! We've been hard at work these past few months implementing improvements to multiple databases, adding support for destroying associations in jobs instead of in memory, turning errors into objects, and so much more.
+
+It's amazing how Rails has grown over the years and while we have some improvements to make to the [onboarding process](https://weblog.rubyonrails.org/2020/5/7/A-May-of-WTFs/), Rails has never been better. The features in this release focus on adding the functionality you need to keep your application up and running for years to come.
+
+Let's look at some of the new functionality:
+
+## Multi-DB Improvements
+
+### Per-database Connection Switching
+
+Rails 6.1 provides you with the ability to [switch connections per-database](https://github.com/rails/rails/pull/40370). In 6.0 if you switched to the `reading` role then all database connections also switched to the reading role. Now in 6.1 if you set `legacy_connection_handling` to `false` in your configuration, Rails will allow you to switch connections for a single database by calling `connected_to` on the corresponding abstract class.
+
+### Horizontal Sharding
+
+Rails 6.0 provided the ability to functionally partition (multiple partitions, different schemas) your database but wasn't able to support horizontal sharding (same schema, multiple partitions). Rails wasn't able to support horizonal sharding because models in Active Record could only have one connection per-role per-class. This is now fixed and [horizontal sharding](https://github.com/rails/rails/pull/38531) with Rails is available.
```

Comment:
```suggestion
Rails 6.0 provided the ability to functionally partition (multiple partitions, different schemas) your database but wasn't able to support horizontal sharding (same schema, multiple partitions). Rails wasn't able to support horizontal sharding because models in Active Record could only have one connection per-role per-class. This is now fixed and [horizontal sharding](https://github.com/rails/rails/pull/38531) with Rails is available.
```